File: 1° semestre/APC verão/Raylan da Silva Sales_lab_4/Python/Lab4b.py
def bubble_sort(lista):
    elementos = len(lista)-1
    ordenado = False
    while not ordenado:
        ordenado = True
        for i in range(elementos):
            if lista[i] > lista[i+1]:
                lista[i], lista[i+1] = lista[i+1],lista[i]
                ordenado = False
    return lista

qtd = 0
vetor=[]
vetor2=[1000]
entrada = 0
flag = 1
flag2 = 0

while entrada!=11:
    entrada = float(input("Digite a entrada: "))
    if entrada==11:
        break
    else:
        qtd+=1
        vetor.append(qtd)
        vetor.insert(qtd, entrada)

vetor = bubble_sort(vetor)

if qtd % 2 == 0:
    qtd = qtd/2
    resultado = (vetor[qtd]+vetor[qtd+1])/2
    print("\nO conjunto eh par.\nMediana: {}".format(resultado))
else:
    qtd/2
    qtd+=1
    resultado = vetor[qtd]
    print("\nO conjunto eh impar.\nMediana: {}".format(resultado))

File: 1° semestre/APC verão/Raylan da Silva Sales_lab_4/C/Lab4b.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main() {
    int qtd=0, k, j;
    float resultado, *vetor, aux, entrada, vetor2[1000];

    do{
        printf("Digite o numero: ");
        scanf("%f",&entrada);
        if(entrada==11){
            break;
        }else{
            qtd++;
            vetor2[qtd]=entrada;
        }
    }while(entrada!=11);

    vetor = (float*)malloc(qtd*sizeof(float));
    for(int cont = 0;cont<qtd;cont++){
        vetor[cont] = vetor2[cont];
    }
    for (k = 1; k <qtd; k++) {
        for (j = 0; j < qtd - 1; j++) {
            if (vetor[j] > vetor[j + 1]) {
                aux = vetor[j];
                vetor[j] = vetor[j + 1];
                vetor[j + 1] = aux;
            }
        }
    }

    if(qtd % 2 == 0){
        qtd = qtd/2;
        resultado = (vetor[qtd] + vetor[qtd+1])/2;
        printf("\nO conjunto eh par.\nMediana: %.1f",resultado);
    }else{
        qtd = qtd/2;
        qtd++;
        resultado = vetor[qtd];
        printf("\nO conjunto eh impar.\nMediana: %.1f",resultado);
    }
    free(vetor);
    return 0;
}
